src/spaniel-observer.ts

```diff
@@ -129,8 +138,11 @@ export class SpanielObserver {
     }
   }
   private flushQueuedEntries() {
-    this.callback(this.queuedEntries.slice());
-    this.queuedEntries = [];
+    let queued = this.queuedEntries.slice();
```

Comment:
why copy and allocate a new?  why not just flush and reset? what happens if something in the queue errors?